src/sentry/monitors/endpoints/organization_monitor_checkin_attachment.py

```diff
@@ -4,20 +4,30 @@
 from rest_framework.request import Request
 from rest_framework.response import Response
 
-from sentry.api.authentication import DSNAuthentication
 from sentry.api.base import region_silo_endpoint
+from sentry.api.endpoints.event_attachment_details import EventAttachmentDetailsPermission
 from sentry.models import File
 
-from .base import MonitorCheckInAttachmentPermission, MonitorCheckInEndpoint
+from .base import MonitorCheckInEndpoint, ProjectMonitorPermission
+
+
+class MonitorCheckInAttachmentPermission(EventAttachmentDetailsPermission):
+    scope_map = ProjectMonitorPermission.scope_map
+
+    def has_object_permission(self, request: Request, view, project):
+        result = super().has_object_permission(request, view, project)
+
+        # Allow attachment uploads via DSN
+        if request.method == "POST":
+            return True
```

Comment:
@davidenwang Actually won't this always be false now? We can just drop here huh?